Added SearchBar component

mkaizen opened the pull request:
Added the SearchBar component and implemented it in the homepage
![image](https://github.com/user-attachments/assets/68fdef82-a6c0-4005-b314-87e86c86fc47)

mkaizen committed:
Added SearchBar component

Added the SearchBar component and implemented it in the homepage

mkaizen committed:
Fixed search bar

Didn't realize project was nested in open-recipes folder and had some issues with dependencies. working searchbar component example in index.tsx

matthew-latanafrancia commented on open-recipes/src/app/(tabs)/index.tsx:
```diff
@@ -1,20 +1,24 @@
       headerImage={
         <Image
-          source={require('@/assets/images/partial-react-logo.png')}
+          source={require('@/assets/images/partial-react-logo.png')} // Fixed: Added require
```
we can remove this comment

matthew-latanafrancia commented on open-recipes/src/app/(tabs)/index.tsx:
```diff
@@ -67,4 +71,8 @@ const styles = StyleSheet.create({
     position: 'absolute',
   },
-});
+  container: {
```
i forget if this is a thing in react native because its been awhile.  is it common to have stylesheets in like a separate file for project structure? cc: @qkstngus000

qkstngus000 replied:
Yeah now we have added tailwind in our project, we should first re-factor all our style codes around it, and any other custom css, we probably should put it in global.css page or in prop styles if it is quick thing.

mkaizen committed:
feat: Add Flipp API integration and display results in search bar

Added an API endpoint (flippApi.ts) and implemented a search result list formatted using (itemGrid.tsx) under the search bar to display results. Removed boilerplate app files.

matthew-latanafrancia commented on open-recipes/package.json:
```diff
@@ -15,38 +15,40 @@
+    "open-recipes": "file:",
     "react": "18.2.0",
     "react-dom": "18.2.0",
-    "react-native": "0.74.5",
```
Do you know why some of these package versions are getting downgraded?  Did you need to make the package.json changes?

mkaizen replied:
I think it was from running `npx expo- doctor` there was an issue with compatibility between dependencies
